controllers/appvolumereplication_controller.go

```diff
@@ -0,0 +1,377 @@
+	if err != nil {
+		return "", "", fmt.Errorf("failed to retrieve placementRule using placementRef %s/%s", plRef.Namespace, plRef.Name)
+	}
+
+	return r.extractHomeClusterAndPeerCluster(subscription, placementRule)
+}
+
+func (r *AppVolumeReplicationReconciler) extractHomeClusterAndPeerCluster(
+	subscription *subv1.Subscription, placementRule *plrv1.PlacementRule) (string, string, error) {
+	subStatuses := subscription.Status.Statuses
+
+	const clusterCount = 2
+
+	// decisions := placementRule.Status.Decisions
+	clmap, err := utils.PlaceByGenericPlacmentFields(
+		r.Client, placementRule.Spec.GenericPlacementFields, nil, placementRule)
+	if err != nil {
+		return "", "", fmt.Errorf("failed to get cluster map for placement %s error: %w", placementRule.Name, err)
+	}
+
+	if len(clmap) != clusterCount {
+		return "", "", fmt.Errorf("PlacementRule %s should have made 2 decisions. Found %d", placementRule.Name, len(clmap))
+	}
+
+	idx := 0
+
+	clusters := make([]spokeClusterV1.ManagedCluster, clusterCount)
+	for _, c := range clmap {
+		clusters[idx] = *c
+		idx++
+	}
+
+	d1 := clusters[0]
+	d2 := clusters[1]
+
+	var homeCluster string
+
+	var peerCluster string
+
+	switch {
+	case subStatuses[d1.Name] != nil:
+		homeCluster = d1.Name
+		peerCluster = d2.Name
+	case subStatuses[d2.Name] != nil:
+		homeCluster = d2.Name
+		peerCluster = d1.Name
+	default:
+		return "", "", fmt.Errorf("mismatch between placementRule %s decisions and subscription %s statuses",
+			placementRule.Name, subscription.Name)
+	}
+
+	return homeCluster, peerCluster, nil
+}
+
+func (r *AppVolumeReplicationReconciler) doCreateOrUpdateManifestWork(name string, namespace string,
+	homeCluster string, peerCluster string) error {
+	r.Log.Info("attempt to create and update ManifestWork")
+
+	manifestWork := r.createManifestWork(name, namespace, homeCluster, peerCluster)
+
+	found := &ocmworkv1.ManifestWork{}
+
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: manifestWork.Name, Namespace: homeCluster}, found)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			r.Log.Error(err, "Could not fetch ManifestWork")
+
+			return moreerrors.Wrap(err, "failed to fetch manifestWork")
+		}
+
+		r.Log.Info("Creating ManifestWork", "ManifestWork", manifestWork)
+
+		return r.Client.Create(context.TODO(), manifestWork)
+	} else if !reflect.DeepEqual(found.Spec, manifestWork.Spec) {
+		manifestWork.Spec.DeepCopyInto(&found.Spec)
+
+		r.Log.Info("Updating ManifestWork", "ManifestWork", manifestWork)
+
+		return r.Client.Update(context.TODO(), found)
+	}
+
+	return nil
+}
+
+func (r *AppVolumeReplicationReconciler) createManifestWork(name string, namespace string,
+	homeCluster string, peerCluster string) *ocmworkv1.ManifestWork {
+	vrgClientManifest := r.createVRGClientManifest(name, namespace, homeCluster, peerCluster)
+
+	manifestwork := &ocmworkv1.ManifestWork{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf(manifestWorkNameFormat, homeCluster),
```

Comment:
Ok naming again :)

So if we have 2 different manifest work for 2 different AVRs the names within the managedcluster namespace on the hub will collide if we use `<managedcluster>-vrg-manifestwork`, afaict.

Should we make this, `<vrg-name>-<vrg-namespace>-vrg-mw` instead?